### ISSUE TITLE
private deploy

### ISSUE THREAD

**Issue by vks2** [MangoInstantMessenger/MangoMessengerAPI#504]
Is it possible to deploy it on a private server, not azure without the need of breaking down whole codebase?

**Comment by kolosovpetro** [comment]
Yeah, it can be easily deployed on virtual machines both Cloud IaaS or On-premises using

- Docker compose
- Build and deploy self contained artifact

Deploy can be done via SSH or Ansible + CI/CD tool like Github Actions or Azure pipelines etc

Note that it is vendor locked to use Azure Blob storage at the moment, to avoid it simply implement `IBlobService` to use AWS S3 or another file storage solution

**Comment by vks2** [comment]
could you provide an example in separate branch? we're interested in on-premise deploy. CI-CD is also problematic due to corp restrictions
PS are you free to provide some small consulting project?

**Comment by kolosovpetro** [comment]
I am not sure I have enough time for something like consulting now
